src/syscalls/native.rs

```diff
@@ -590,14 +590,53 @@ pub struct SpawnArgs {
 /// The parent process calls the Spawn system call, which creates a new process (a child process) that is an
 /// independent ckb-vm instance. It's important to note that the parent process will not be blocked by the child
 /// process as a result of this syscall.
-/// Note: available after ckb2023.
+/// Note: available after ckb 2nd hardfork.
 ///
 /// # Arguments
 ///
 /// * `index`, `source`, `bounds` and `place` - same as exec.
 /// * `spgs` - spawn arguments.
 ///
 /// Returns success or a syscall error.
+///
+/// # Scheduler Algorithm
+///
+/// This document describes the design and functionality of a scheduler algorithm, covering process states, system call
+/// behavior, message handling, and priority rules. The scheduler manages virtual processes, transitioning them through
+/// various states based on operations and interactions, to ensure efficient scheduling.
+///
+/// # Thread States
+///
+/// Each process within this scheduler has one of the following five states:
```

Comment:
Just throwing something out: is `Running` also a state?

Reply:
From the code, it seems that `Running` should not be a state

https://github.com/nervosnetwork/ckb/blob/develop/script/src/types.rs#L394-L410

Reply:
Yes but the code in CKB is the actual implementation, the `VmState` structure in implementation, does not always mean a conceptual model. I'm raising a different question: from a higher level, when one is using `ckb-std` to construct apps on CKB, is running, treated as a state of a program?

Looking at literature of OSes: running is treated as a state: https://www.geeksforgeeks.org/states-of-a-process-in-operating-systems/

Reply:
Added the Running state, and added information on how to enter and exit the Running state at the end.